runtime/compiler/env/j9method.cpp

```diff
@@ -7017,12 +7042,26 @@ TR_ResolvedJ9Method::getResolvedDynamicMethod(TR::Compilation * comp, I_32 callS
 
          {
          TR::VMAccessCriticalSection getResolvedDynamicMethod(fej9());
-         targetJ9MethodBlock = fej9()->targetMethodFromMemberName((uintptr_t) fej9()->getReferenceElement(*invokeCacheArray, JSR292_invokeCacheArrayMemberNameIndex)); // this will not work in AOT or JITServer
+         targetJ9MethodBlock = fej9()->targetMethodFromMemberName((uintptr_t) fej9()->getReferenceElement(*invokeCacheArray, JSR292_invokeCacheArrayMemberNameIndex));
```

Comment:
This line dereferences `invokeCacheArray` to get the array reference, then dereferences that to get a `MemberName` reference from the element, and finally dereferences *that* to get `vmtarget`

The latter two dereferences definitely can't happen on the server, and I'm pretty sure that `*invokeCacheArray` can't either

Similarly in `getResolvedHandleMethod()`

Reply:
😞